Server/ConcordiaCurriculumManager/Controllers/DossierController.cs

```diff
@@ -56,21 +53,21 @@ public async Task<ActionResult> GetDossiersByID()
         }
     }
 
-    [HttpGet("{id}")]
+    [HttpGet("{dossierId}")]
     [SwaggerResponse(StatusCodes.Status200OK, "Dossier retrieved successfully", typeof(DossierDetailsDTO))]
     [SwaggerResponse(StatusCodes.Status404NotFound, "Dossier not found")]
     [SwaggerResponse(StatusCodes.Status500InternalServerError, "Unexpected error")]
-    public async Task<IActionResult> GetDossierByDossierId(Guid id)
+    public async Task<IActionResult> GetDossierByDossierId([FromRoute, Required] Guid dossierId)
     {
         try
         {
-            var dossier = await _dossierService.GetDossierDetailsById(id);
+            var dossier = await _dossierService.GetDossierDetailsById(dossierId);
             var dossierDetails = _mapper.Map<DossierDetailsDTO>(dossier);

```

Comment:
What happens if the dossier is null? Does this throw an exception?

Reply:
It does not throw an error, it returns a 204 if the dossier is null. This is the original behaviour; I did not change it.